PollBuddy-Server/backend/modules/utils.js

```diff
@@ -0,0 +1,35 @@
+const mongoConnection = require("../modules/mongoConnection.js");
+
+
+// Helper function for creating specified http response (https://pollbuddy.app/api/users)
+// error and data are optional
+function createResponse(isSuccess = true, error, data) {
```

Comment:
Can I request a slight change here? I feel we could save a bit of complexity.

Something like `function createResponse(data, error=null) { ... }` would then make it a bit more easy to use, allowing both `createResponse({"somedata": "toreturn"});` and `createResponse({"somedata": "toreturn"}, "Plus an error message");`. Then all you have to pass in is `data`, and optionally an `error` string at the end. I think your method requires passing in an "error" even if one doesn't exist, plus `isSuccess` is a bit redundant as the presence of `error` could then be used to determine the status. 

Honestly, I'm probably being overly concerned here and I apologize for that. Feel free to not do it lol but I think it'd make things a lot easier in the future as then all someone would have to think about is passing in the return data and optionally an error string and the function will do the rest.

Reply:
(Also if on error someone doesn't want to return any data, they can just pass in `{}`". Might not be a bad idea anyways as then we can guarantee the existence of data in backend endpoints, which may be desirable for consistency.

Reply:
Absolutely, now that I've look at it again it's actually horrible code design lol.

Reply:
Lol well I definitely wouldn't say that, just some stuff that could be done differently is all. No worries :)